xdsl/irdl/constraints.py

```diff
@@ -1,14 +1,15 @@
 from __future__ import annotations
 
+import abc
 from abc import ABC, abstractmethod
-from collections.abc import Sequence
+from collections.abc import Sequence, Set
```

Comment:
Is there a reason you are using `Set` instead of `set`?
AFAIK, we should always prefer `set` since Python 3.something?

Reply:
`Set` is more general and in particular includes `dict.keys()`. Note that `collections.abc.Set =/= typing.Set`

Reply:
Oh god, I did not know about `typing.Set` versus `collections.abc.Set`...

Reply:
Makes autocompletion very annoying, especially since `typing.Set` is useless since Python 3.something as you said

Reply:
Also Set and Dict are immutable collections if I remember correctly, it also helps with variance things